src/com/pentalog/pentastagiu/homework2/Basket.java

```diff
@@ -0,0 +1,47 @@
+package com.pentalog.pentastagiu.homework2;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Basket {
+
+    private Integer bascketId;
+    private Double totalPrice;
+    private Customer customer;
+    private List<Product> products = new ArrayList<>();
+
+    public Double getTotalPrice() {
+        return totalPrice;
+    }
+
+    public List<Product> getProducts() {
+        return products;
+    }
+
+    public void setTotalPrice(Double totalPrice) {
+        this.totalPrice = totalPrice;
+    }
+
+    //Every client has associated an basket where he is adding the products
+    //He can buy one by one
+    public void add(Product product){
+        getProducts().add(product);
```

Comment:
vad ca ai folosit getteri/setteri in metoda asta; nu este nevoie, te poti folosi direct de variabilele private tinand cont ca ai access la ele din aceeasi clasa

da, este o buna practica sa folosesti fielduri pe care le declari private si getteri/setteri, dar in general ti-as sugera sa incepi fara getteri/setteri si sa ii adaugi doar in momentul in care ai nevoie
cand scriem cod e ok sa dam acces la cat mai putin (mai ales la setteri, pentru a evita ca cineva sa ii foloseasca intr-un mod gresit: de ex cineva ar putea sa vina si sa foloseasca metoda setTotalPrice si sa iti altereze pretul calculat de tine adaugand produs cu produs)